R/zzz.R

```diff
@@ -61,9 +61,61 @@
   s3_register <- asNamespace("rlang")$s3_register
   s3_register("pillar::type_sum", "python.builtin.object")
 
+  if (is_positron()) {
+
+    setHook("reticulate.onPyInit", function() {
+
+      # options("ark.testing" = TRUE)
+      inspectors <- import_positron_ipykernel_inspectors()
+      if(is.null(inspectors)) {
+        # warning("positron_ipykernel.inspectors could not be found, variables pane support for Python objects will be limited")
+        return()
+      }
+
+      .globals$get_positron_variable_inspector <- inspectors$get_inspector
+
+      .ark.register_method <- get(".ark.register_method", envir = globalenv())
+
+      for (method_name in c(
+        "ark_positron_variable_display_value",
+        "ark_positron_variable_display_type",
+        "ark_positron_variable_has_children",
+        "ark_positron_variable_kind",
+        "ark_positron_variable_get_child_at",
+        "ark_positron_variable_get_children"
+      )) {
+        method_sym <- as.name(paste0(method_name, ".python.builtin.object"))
+        .ark.register_method(
+          method_name, "python.builtin.object",
+          call(":::", quote(reticulate), method_sym))
+      }
+    })
+  }
 }
 
 
 # .onUnload <- function(libpath) {
 #   py_finalize() # called from reg.finalizer(.globals) instead.
 # }
+
+
+import_positron_ipykernel_inspectors <- function() {
+  if(!is_positron())
+    return (NULL)
+
+  x <- Sys.getenv("_")
```

Comment:
@dfalbel (@lionel-?) Is there a better way to get the path to the Positron installation from R? I'm not sure if this is the most reliable way to resolve the path to the `inspectors.py` file that ships with Positron.

Reply:
@lionel- We'll open a PR to Ark, exporting a function like `.ps.positron_ipykernel_path`, to return the path to the `positron_ipykernel` python module that ships with Positron.

Reply:
yes that would be reasonable. `.ps` stands for positron so `.ps_ipykernel_path`?

This function could query the positron-r extension for that path via UIComm RPC.